nx_parallel/utils/chunk.py

```diff
@@ -1,66 +1,99 @@
 import itertools
 import os
+
 import networkx as nx
 
+from nx_parallel.utils.types import GraphIteratorType
+
+
 __all__ = ["chunks", "get_n_jobs", "create_iterables"]
 
 
-def chunks(iterable, n):
-    """Divides an iterable into chunks of size n"""
+def chunks(iterable, n_chunks):
+    """Yield exactly `n_chunks` chunks from `iterable`, balancing the chunk sizes."""
+    iterable = list(iterable)
+    k, m = divmod(len(iterable), n_chunks)
     it = iter(iterable)
-    while True:
-        x = tuple(itertools.islice(it, n))
-        if not x:
-            return
-        yield x
+    for _ in range(n_chunks):
+        chunk_size = k + (1 if m > 0 else 0)
+        m -= 1
+        yield tuple(itertools.islice(it, chunk_size))
```

Comment:
The current behaviour of this function is changed here. Was that intended by you or is this a bug?

```py
>>> data = list(range(11))
>>> list(new_chunks(data, 4))
[(0, 1, 2), (3, 4, 5), (6, 7, 8), (9, 10)]
>>> list(old_chunks(data, 4))
[(0, 1, 2, 3), (4, 5, 6, 7), (8, 9, 10)]
```

Reply:
In today's discussion, @dPys and I figured that the docstrings indeed says that the `new_chunk` is the way we chunk, so this is more of a bug getting fixed. 

@dschult do you have any thoughts on what should be the default chunking - making chunks of size `n` or `n` number of chunk? (We do pass `n_jobs` as `n` in the chunking function, so it does make sense to have `n` chunks instead of chunks of size `n`). Also it would be great if you could give a final review as this PR is almost ready to be merged! Thank you :)

Reply:
<img width="1392" alt="Screenshot 2024-09-20 at 2 48 31 PM" src="https://github.com/user-attachments/assets/68780869-9f33-42c2-a8cd-6022d3531d5f">
The above plot is for the new chunking -- runtime does not change as a function of graph size in the case of betweenness centrality. 

I had trouble even getting the old (i.e. current) chunking to work consistently across all configurations of chunks / n_cores, otherwise I'd post its benchmarks too. Can try to see if I can figure out why...

Reply:
I think n-chunks is what we should be providing (to match n_jobs). I think it doesn't really matter what size the chunks are -- bigger will just take longer.  It is the number of chunks that is restricted. So that's a go from my perspective.

Nice plots of time vs number of nodes!  (note that graph "size" is technically the number of edges. graph "order" is the number of nodes, but "number of nodes" seems to be the term most people use.)  
It looks good up to N=250. But that's not actually that big. Still, it shows the trends that we want. And it is clear that we need N>250 for parallel to pay off in execution time in this situation. The memory plot seems to suggest that the number of cores doesn't affect memory. 

Maybe these plots should go in a main comment so it doesn't get "hidden" when the discussion is "resolved". :)